src/gui/properties/speedplotview.h

```diff
@@ -76,14 +76,30 @@ class SpeedPlotView: public QGraphicsView
     void setGraphEnable(GraphID id, bool enable);
     void setViewableLastPoints(TimePeriod period);
 
-    void pushPoint(PointData point);
+    void pushPoint(const PointData &point);
 
     void replot();
 
 protected:
     virtual void paintEvent(QPaintEvent *event);
 
 private:
+    typedef boost::circular_buffer<PointData> PointBufferType;
+    class Divider{
+    public:
+        explicit Divider(int _N,PointBufferType* _output)
+            : counter(-1)
+            , N(_N)
+            , output(_output)
+        {}
+        void push(const PointData &point);
+        int counter;
```

Comment:
Move it to private section and implement appropriate method with meaningful name, e.g.:
```c++
bool done() const { return counter == 0; }
```

Reply:
did so.